src/main/target/SITL/target.h

```diff
@@ -0,0 +1,234 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+// SITL (software in the loop) simulator
+#define TARGET_BOARD_IDENTIFIER "SITL"
+
+// use simulatior's attitude directly
+#define SKIP_IMU_CALC
+
+// file name to save config
+#define EEPROM_FILENAME "eeprom.bin"
+
+#define SIMULATOR_BUILD
+
+#define U_ID_0 0
+#define U_ID_1 1
+#define U_ID_2 2
+
+#define ACC
+#define USE_FAKE_ACC
+
+#define GYRO
+#define USE_FAKE_GYRO
+
+#define MAG
+#define USE_FAKE_MAG
+
+#define BARO
+#define USE_FAKE_BARO
+
+#define USABLE_TIMER_CHANNEL_COUNT 0
+
+#define USE_UART1
+#define USE_UART2
+#define USE_UART3
+#define USE_UART4
+#define USE_UART5
+#define USE_UART6
+#define USE_UART7
+#define USE_UART8
+
+//#define USE_SOFTSERIAL1
+//#define USE_SOFTSERIAL2
+
+#define SERIAL_PORT_COUNT 8
+
+#define DEFAULT_RX_FEATURE      FEATURE_RX_MSP
+#define DEFAULT_FEATURES        (FEATURE_GPS | FEATURE_TELEMETRY)
+
+#define USE_PARAMETER_GROUPS
+
+#undef STACK_CHECK // I think SITL don't need this
+#undef USE_DASHBOARD
+#undef TELEMETRY_LTM
+#undef USE_ADC
+#undef USE_VCP
+#undef USE_PPM
+#undef USE_PWM
+#undef SERIAL_RX
+#undef USE_SERIALRX_CRSF
+#undef USE_SERIALRX_IBUS
+#undef USE_SERIALRX_SBUS
+#undef USE_SERIALRX_SPEKTRUM
+#undef USE_SERIALRX_SUMD
+#undef USE_SERIALRX_SUMH
+#undef USE_SERIALRX_XBUS
+#undef LED_STRIP
+#undef TELEMETRY_FRSKY
+#undef TELEMETRY_HOTT
+#undef TELEMETRY_SMARTPORT
+#undef TELEMETRY_MAVLINK
+#undef USE_RESOURCE_MGMT
+#undef CMS
+#undef TELEMETRY_CRSF
+#undef TELEMETRY_IBUS
+#undef TELEMETRY_JETIEXBUS
+#undef TELEMETRY_SRXL
+#undef USE_SERIALRX_JETIEXBUS
+#undef VTX_COMMON
+#undef VTX_CONTROL
+#undef VTX_SMARTAUDIO
+#undef VTX_TRAMP
+
+#undef USE_I2C
+#undef USE_SPI
+
+#define FLASH_SIZE 2048
+
+# define DEFIO_PORT_USED_COUNT 0
```

Comment:
id_def_generated.h should calculate this from TARGET_IO_PORTx
IO is expected to work in unittests, so it should be easy to adapt it for simulator